app/Parsers/RequestHeaderParser/RequestHeaderParser.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace App\Parsers;
+
+use App\Parsers\RequestHeaderParser\Entities\Header;
+use App\Parsers\RequestHeaderParser\Entities\Response;
+use App\Parsers\Exceptions\FailedFetchingResponseHeadersException;
+
+class RequestHeaderParser
+{
+    /**
+     * Allowed methods.
+     *
+     * @var array
+     */
+    private static $allowedMethods = [
+        'GET',
+        'HEAD',
+        'POST',
+        'PUT',
+        'DELETE',
+        'CONNECT',
+        'OPTIONS',
+        'TRACE',
+        'PATCH',
+    ];
+
+    /**
+     * Method that should be used to execute the HTTP request.
+     *
+     * @var string
+     */
+    private $method;
+
+    /**
+     * Url that should be used to execute the HTTP request.
+     *
+     * @var string
+     */
+    private $url;
+
+    /**
+     * The response headers from the HTTP request.
+     *
+     * @var array
+     */
+    private $headers;
+
+    /**
+     * @param $method
+     * @throws InvalidEndpointMethodException
+     */
+    public function setMethod($method)
+    {
+        if (! in_array($method, static::$allowedMethods)) {
+            throw new InvalidEndpointMethodException();
+        }
+
+        $this->method = $method;
+    }
+
+    /**
+     * @param $url
+     * @throws InvalidEndpointUrlException
+     */
+    public function setUrl($url)
+    {
+        if (! filter_var($url, FILTER_VALIDATE_URL)) {
+            throw new InvalidEndpointUrlException();
+        }
+
+        $this->url = $url;
+    }
+
+    /**
+     * Execute request and parse responses with headers.
+     *
+     * @return Response[]
+     */
+    public function getResponses(): array
+    {
+        $this->execute();
+
+        /* @var Response[] $responses */
+        $responses = [];
```

Comment:
Why not using a `Collection`?

Reply:
The RequestHeaderParser is a class that can be used in multiple projects/environment/packages. It should be nice if collections are not mandatory.